#!/usr/bin/env Rscript

######  
suppressMessages(library("docopt"))
"
Usage:
	filter_SNPs_and_create_whitelist_for_populations.R --vcf.file=<vcf.file> --white.list.out=<white.list.out> --plot.out=<plot.out> --desc=<desc> 

Description: This script will create a white-list of SNPs for populations to run.  

Options:
	--vcf.file=<vcf.file>				 	from populations module
	--white.list.out=<white.list.out>		path of where to save list of SNPs to include in populations analysis
	--plot.out=<plot.out>					path to save plots to
	--desc=<desc>							description of analysis (eg: m10)
" -> doc
######

###### PARAMETERS ##########
# Set the parameters:
today <- Sys.Date()											# Set the date that will go on the end of the files generated by this script
today <- format(today, format="%m%d%y")
#############################



##### Load arguments:
opts <- docopt(doc)
#print(opts)
vcf.file <- opts$vcf.file
white.list.out <- opts$white.list.out
plot.out <- opts$plot.out
mydesc <- opts$desc


# vcf.file <- "../data/STACKS_processed/7_depth_optimization/m10/rxstacks_corrected/batch_4.vcf"
#white.list.out <- "../data/STACKS_processed/7_depth_optimization/m10/rxstacks_corrected/"
#plot.out <- "../results/5_optimizing_depth/"
#mydesc <- "m10"



##### Read in VCF file:
print("reading in VCF file")
vcf <- read.table(vcf.file, sep="\t", header=TRUE, stringsAsFactors=FALSE, skip = 9, comment.char="")



##### Filter out loci with 2 x SD(mean coverage):
# Pull out number of reads per locus for each sample
coverage <- apply(vcf[,10:dim(vcf)[2]], c(1,2), function(x) {strsplit(x, split=":")[[1]][2]})

# Replace zeros with NAs
coverage[coverage == 0] <- NA

# Calculate the mean read depth for individual where a SNP was called
mean.coverage <- apply(coverage, 1, function(x) {mean(as.numeric(x), na.rm=TRUE)})

# What is the mean read depth?
mean.of.the.mean <- mean(mean.coverage)
sd.of.the.mean <- sd(mean.coverage)

# Which SNPs are 2 x SD(mean coverage)?
high.coverage.sites <- which(mean.coverage >= (mean.of.the.mean + 2*sd.of.the.mean))

filtered.coverage <- coverage[-high.coverage.sites,]



##### Stats on filtered loci:
mean.filtered.coverage <- apply(filtered.coverage, 1, function(x) {mean(as.numeric(x), na.rm=TRUE)})
mean.of.the.filtered <- mean(mean.filtered.coverage)



##### Plots before and after filtering:
print("plotting average coverage distribution")

pdf(paste0(plot.out, "hist_coverage_before_and_after_filtering_", mydesc,"_",today,"ERD.pdf"), width=3.5, height=9)
par(mfrow=c(3, 1))
# Coverage before filtering:
hist(mean.coverage, main=paste(mydesc, ": average coverage per loci"), xlab="reads", col="gray")
abline(v=mean.of.the.mean, col="red")
mtext(text=round(mean.of.the.mean,2), side=3, line=0, outer=FALSE, at=mean.of.the.mean, col="red")

# Log10 coverage before filtering:
hist(log10(mean.coverage), main=paste(mydesc, ": average coverage per loci (log10)"), xlab="log10(reads)", col="gray")
abline(v=log10(mean.of.the.mean), col="red")
mtext(text=round(log10(mean.of.the.mean),2), side=3, line=0, outer=FALSE, at=log10(mean.of.the.mean), col="red")

# Coverage after filtering
hist(mean.filtered.coverage, main=paste(mydesc, ": average coverage per loci after filtering"), xlab="reads", col="gray")
abline(v=mean.of.the.filtered, col="red")
mtext(text=round(mean.of.the.filtered, 2), side=3, line=0, outer=FALSE, at=mean.of.the.filtered, col="red")
dev.off()


##### Create a whitelist:
whitelist.vcf <- vcf[-high.coverage.sites,]

write.table(whitelist.vcf$ID, paste0(white.list.out, "whitelist_loci_after_coverage_filtering.txt"), sep="\n", row.names=FALSE, col.names=FALSE, quote=FALSE)

print("DONE!")
