#!/usr/bin/env Rscript

###### 
# This script will output a table of the number of reads put into mapping, are mapped, that don't map, and that don't map because of the ambiguity threshold for each sample (to mtDNA). 
# ./summarize_mapped_reads_after_bowtie_mtDNA.R 
# input: 	log files from bowtie mapping
# output: 	text file containing the mapping info
######

###### PARAMETERS ##########
# Set the parameters:
today <- Sys.Date()											# Set the date that will go on the end of the files generated by this script
today <- format(today, format="%m%d%y")
#############################



##### Load libraries:
suppressMessages(library("dplyr"))



##### Get list of files:
path <- "data/STACKS_processed/3_aligned_to_mitochondrial_genome/"
files <- list.files(path=path, pattern=".log")



##### Read in file and pull information:
parse_read_info <- function(filename) {
	mydata <- read.table(file=paste0(path, filename), sep="\n", header=FALSE, stringsAsFactors=FALSE, comment.char="")$V1
	rp <- as.numeric(strsplit(mydata[1], split=": ")[[1]][2])
	ra <- as.numeric(strsplit(mydata[2], split=" ")[[1]][9])
	rfa<- as.numeric(strsplit(mydata[3], split=" ")[[1]][7])
	myreads <- c(rp, ra, rfa)
}

reads <- t(sapply(files, parse_read_info))
rownames(reads) <- gsub(".fastq.log", "", row.names(reads))
colnames(reads) <- c("reads_processed", "reads_aligned", "reads_failing_alignment")

write.table(reads, paste0("results/2_processing_info/table_alignment_stats_bowtie_mtDNA_",today,"ERD.txt"), sep="\t", quote=FALSE)



print("DONE!")
